Makefile

```diff
@@ -1,26 +1,22 @@
-TEST_TARGETS=zhenech_DASH_katello-client\:el5 zhenech_DASH_katello-client\:el6 zhenech_DASH_katello-client\:el7
 USE_SELINUX=$(shell test -d /sys/fs/selinux && echo ":Z")
+DOCKERFILE  ?= Dockerfile.el7
+PYTHON ?= python2
+
+install:
+	cd src/ && $(PYTHON) setup.py install
 
 test-python3:
 	pip3 install -r test-requirements.txt --user
-	python3 -m unittest discover -s test/
+	python3 -m unittest discover -v -s test/
 
 test-python2:
 	pip install -r test-requirements.txt --user
-	python2 test/python2_suite.py
+	python -m unittest discover -v -s test/
 
-test: $(addprefix test-,$(TEST_TARGETS))
-	flake8 --ignore E501 ./bin/* src/ || true
+docker-run:
+	docker build -f $(CURDIR)/docker/$(DOCKERFILE) -t kht-builder .
+	docker run -it --volume $(CURDIR):/app$(USE_SELINUX) --workdir=/app kht-builder /bin/bash
 
-test-%:
-	docker run -it --volume $(CURDIR):/app$(USE_SELINUX) --workdir=/app $(subst _DASH_,/,$*) python -m compileall src/
-	docker run -it --volume $(CURDIR):/app$(USE_SELINUX) --env="PYTHONPATH=src/:src/yum-plugins/" \
-		--workdir=/app $(subst _DASH_,/,$*) \
-		bash -c \
-			'for binary in bin/*; do \
-				if [ "$${binary}" = "bin/katello-tracer-upload" ]; then \
-					continue; \
-				fi; \
-				python -- $${binary} -f ; \
-				python -- $${binary} -help ; \
-			done'
+docker-test:
+	docker build -t kht-builder -f $(CURDIR)/docker/$(DOCKERFILE) .
+	docker run -it --volume $(CURDIR):/app$(USE_SELINUX) --workdir=/app kht-builder /bin/bash ./test/test_docker.sh
```

Comment:
You could set WORKDIR in the Dockerfile itself and avoid this little nugget. Is the /bin/bash declaration required?

Reply:
You could also consider baking test/test_docker.sh into the image as the CMD that is run by default. If you don't plan to have to iterate much on that script then your image and run declaration is reduced down to mount the proper volume and go.